### ISSUE TITLE
miraclecast build instructions for raspberry pi 4b(8GB ram) version with raspbian OS 

### ISSUE THREAD

**Issue by raghavk92** [albfan/miraclecast#397]
Hi ,
Please if anyone knows how to build and run miraclecast on rasbpian OS buster (latest December edition )

In the wiki section a ready to go image is available for raspberry pi 3 but not 4. And also there are instructions for building and running this on ubuntu 18.04 and 20.04 so if someone can please give a line by line instruction on how to build and run on raspbian OS buster on the raspberry pi 4B ( 8 GB RAM) version and also raspberry pi 4 in general it would be really useful.

Although raspbian OS 64 bit is currently in beta so if possible please provide instructions for the 32 bit and if possible 64 bit as well.

Please reply if possible
thanks & Regards
Raghav

**Comment by albfan** [comment]
Instructions for rapsbian are same as for ubuntu.

This project do not provide precompiled releases, so you should build on machine to have exact arch or learn how to build it multiarch, which is out of scope of this project.

I can provide a tested raspbian image if someone donates me a raspberry pi 4. :wink:

**Comment by raghavk92** [comment]
@albfan  I built and installed it with cmake as in ubuntu . I just wasn't sure if the instructions are same for raspberry pi. i am new to all this. I just have problems running miraclecast now. I don't have network manager so do you want me to install network manager and then stop it . Dont you think if i don't have network manager it really doesn't need to be stopped or installed even either.

I want to use miraclecast with wifi (like with all miracast devices) . You don't loose your internet connection that way. Are the instructions mentioned on readme to do this or does using miraclecast stop my wifi router pairing? 

About donating a raspberry pi i can screen share on discord with you and i can follow your instructions and the image can be created if you wish 

I am using raspbian OS 32 bit buster (latest)

When i run - ./test-hardware-capabilities.sh
wlan0 supports P2P

When I do - "systemctl stop NetworkManager.service" then i get 
Failed to stop NetworkManager.service: Unit NetworkManager.service not loaded.
Do i really need to install network manager to stop it again. Or installation is not needed.

When i do - $ "ps -ef | grep wpa_supplican[t]"
I get below:
------
root       521     1  0 13:09 ?        00:00:02 wpa_supplicant -B -c/etc/wpa_supplicant/wpa_supplicant.conf -iwlan0 -Dnl80211,wext
root      6815  6812  0 15:54 pts/0    00:00:00 /usr/sbin/wpa_supplicant -c /run/miracle/wifi/wlan0-3.conf -C /run/miracle/wifi -i wlan0 -g /run/miracle/wifi/wlan0-3.global
root      6976     1  0 15:57 ?        00:00:00 /sbin/wpa_supplicant -u -s -O /run/wpa_supplicant
-------

when i run - "systemctl stop wpa_supplicant.service"
it works
Then checking with "systemctl status wpa_supplicant.service" I get :
-----
 wpa_supplicant.service - WPA supplicant
   Loaded: loaded (/lib/systemd/system/wpa_supplicant.service; enabled; vendor p
   Active: inactive (dead) since Sat 2020-12-12 17:31:15 IST; 2min 14s ago
  Process: 6976 ExecStart=/sbin/wpa_supplicant -u -s -O /run/wpa_supplicant (cod
 Main PID: 6976 (code=exited, status=0/SUCCESS)

Dec 12 15:57:59 raspberrypi systemd[1]: Starting WPA supplicant...
Dec 12 15:57:59 raspberrypi systemd[1]: Started WPA supplicant.
Dec 12 15:57:59 raspberrypi wpa_supplicant[6976]: Successfully initialized wpa_s
Dec 12 17:31:14 raspberrypi systemd[1]: Stopping WPA supplicant...
Dec 12 17:31:15 raspberrypi systemd[1]: wpa_supplicant.service: Succeeded.
Dec 12 17:31:15 raspberrypi systemd[1]: Stopped WPA supplicant.
lines 1-12/12 (END)
-------
Then running - "ps -ef | grep wpa_supplican[t]"   i get below -
------
root       521     1  0 13:09 ?        00:00:02 wpa_supplicant -B -c/etc/wpa_supplicant/wpa_supplicant.conf -iwlan0 -Dnl80211,wext
root      6815  6812  0 15:54 pts/0    00:00:00 /usr/sbin/wpa_supplicant -c /run/miracle/wifi/wlan0-3.conf -C /run/miracle/wifi -i wlan0 -g /run/miracle/wifi/wlan0-3.global
------

Then running - "sudo miracle-wifid &" a bit earlier i was getting something like -
-------
WARNING: supplicant: wpa_supplicant or driver does not support P2P
-------

Then I  started the wpa_supplicant and stopped again and then did "sudo miracle-wifid &" and now i am getting - 

--------

[2] 7954
pi@raspberrypi:~/Documents/miracle/miraclecast/res $ ERROR: dbus: cannot claim org.freedesktop.miracle.wifi bus-name: -17 (manager_dbus_connect() in /home/pi/Documents/miracle/miraclecast/src/wifi/wifid-dbus.c:941)
---------

Please tell me what should i do .
I also want to use wifi when i am using raspberry pi being the miracast reciever and windows 10 being the sender

Please let me know what to do

**Comment by albfan** [comment]
Is for real that you're suggesting to get my time on a discord channel to help you to create a raspberry pi image? There's a raspberry pi image in this wiki as an example, there's no complexity in create an image, I was just joking about ask for a task that is out of scope of a project, just because you want it.

network manager is just an app that control network. In your case can be other. Whatever it was it spawns wpa_supplicant.

At this time miraclecast cannot ask an existing wpa_supplicant to create interfaces it needs, that's what you should stop whatever is controlling wifi. That's not solved so about help you on that is more, help us on that.

your network manager is starting again wpa_supplicant, that's why you should stop the manager.

**Comment by raghavk92** [comment]
Bro i was joking as well . But i didnt want to offend you.

i dont want a raspberry pi image i have built and installed miraclecast so i just want to know how to run it. 

I gave you all the information that i thought is enough . Let me know how to give you more information. You are saying my network manager is starting wpa_supplicant again but network manager is not installed.

Please tell me how to find my network manager and stop it. But what i also wanted to know does miraclecast let my wifi with internet work together with the casting from one pc (windows 10 ) to the pi?

I am really not getting/understanding what those error messages i specified above completely mean. Please tell me what should i do . I dont know how to provide information to you . Please let me know how

Thanks

**Comment by albfan** [comment]
How to deal with the manager that rules your network is out of scope of this project. Just stop it (go find that info on raspbian docs or other forums) and let only miraclecast to run it.

Just because I reply doesn't mean you can keep asking personal needs here.

I'm trying to show you what you can and what you cannot expect from an issue, and turn all this: "Hey I need help, do this for me" into "how can I help to make this?"

For that, if you find how to stop raspbian wifi manager (however it happens) you can open a chapter on wiki about raspberry pi 4.

If you want miraclecast to work with existing wifi manager, help on #75

**Comment by raghavk92** [comment]
@albfan hi bro , i got it working on the raspberry pi 4 without installing network manager. Need to double check with an ssd that i just ordered because everything just hanged after the cast started so i don't know what the issue was . I ll let you know when i have double checked my method. But it involves disabling just dhcpd instead of network manager

**Comment by albfan** [comment]
That makes sense.

https://raspberrypi.stackexchange.com/questions/74538/how-do-i-configure-dhcpcd-to-call-wpa-supplicant-for-a-specific-interface

Seems like there's a dhcpcd-wlan0 that runs wpa_supplicant, so probably dhcpcd will disable all them.

As said, main objetive is to work in parallel with whatever that runs your network in your system, that this time that involves that you have a recent wpa_supplicant and dbus enabled so miraclecast can create its own interface without affecting the normal ones, but that is still not done.

You can update steps on wiki for raspberry pi 4 when you are sure about the steps.

**Comment by raghavk92** [comment]
sorry i didnt understand what you were saying , i understood that dhcpd should be disabled and the wpa_supplicant service was disabled. Then i ran all the miracle cast steps . It ran fine. As i connected my phone to the miracle wireless display. The gstreamer 1.0 window opened and everything hanged although i was able to see my phone's display on gstreamer but everything hanged . I am not understanding why . Do you know how to troubleshoot this. 

And also I didn't understand what you mean by - "recent wpa_supplicant and dbus enabled so miraclecast can create its own interface without affecting the normal ones, but that is still not done."

sorry i am kind of noob at this but what should i do or steps to run for what you said in the above statement in quotes.

[I even increased the swap size , i thought the ram of the computer would have been full but that didn't stop it from hanging.
Dont know right now how to stop the raspbian os from going into a hang state.]- just some extra information.

**Comment by albfan** [comment]
those instructions wpa_supplicant are for #75 so don't worry.

Please open a proper issue about your hangs using https://github.com/albfan/miraclecast/wiki/FAQ#log-for-support, this has nothing to do with build instructions for raspbian.

**Comment by raghavk92** [comment]
ok i ll check the faq page you linked . But i am able to run miraclecast sometimes on ubuntu 20.10 on rpi4 but it only works with phones. Windows PC's and windows 10 never connect to miraclecast like gstreamer doesn't open. What is the issue here , am i missing something?
